lsp-mode.el

```diff
@@ -5592,6 +5593,9 @@ returned by COMMAND is available via `executable-find'"
                                    :noquery t)))
                        (set-process-query-on-exit-flag proc nil)
                        (set-process-query-on-exit-flag (get-buffer-process stderr-buf) nil)
+                       (with-current-buffer (get-buffer stderr-buf)
+                         ;; Make the *NAME::stderr* buffer buffer-read-only, q to bury, etc.
+                         (special-mode))
```

Comment:
We should check the `major-mode` before activating new `major-mode`. This is done so the initialization logic inside major-mode activation function will not run too much, which can affect perf.
Something along `(unless (eq 'special-mode major-mode) (special-mode))`.

Reply:
Since we are creating the `stderr-buf` we know it won't be in special-mode, thus the condition will always be true, so I don't believe adding an always true condition is needed (or am I missing something an there's a way to have make-process create the buffer in special-mode?).

Reply:
I see, I though that buffer can be set to `(special-mode)` multiple times.